apps/test/unit/lib/levelbuilder/lesson-editor/activitiesEditorReduxTest.js

```diff
@@ -182,100 +183,34 @@ describe('activitiesEditorRedux reducer tests', () => {
 
     it('update activity field', () => {
       let state = reducer(initialState, updateActivityField(1, 'time', 100));
-      assert.deepEqual(
-        [
-          {
-            key: 'x',
-            displayName: 'X',
-            position: 1,
-            time: 100,
-            activitySections: [
-              {key: 'a', position: 1, displayName: 'A'},
-              {key: 'b', position: 2, displayName: 'B'}
-            ]
-          },
-          {
-            key: 'y',
-            displayName: 'Y',
-            position: 2,
-            activitySections: [
-              {key: 'c', position: 1, displayName: 'C'},
-              {key: 'd', position: 2, displayName: 'D'}
-            ]
-          }
-        ],
-        state.activities
-      );
+
+      let expectedState = _.cloneDeep(initialActivities);
+      expectedState[0].time = 100;
+
+      assert.deepEqual(expectedState, state.activities);
     });
 
     it('removes activity', () => {
       let state = reducer(initialState, removeActivity(1));
-      assert.deepEqual(
-        [
-          {
-            key: 'y',
-            displayName: 'Y',
-            position: 1,
-            activitySections: [
-              {key: 'c', position: 1, displayName: 'C'},
-              {key: 'd', position: 2, displayName: 'D'}
-            ]
-          }
-        ],
-        state.activities
-      );
+
+      let expectedState = _.cloneDeep(initialActivities).slice(1);
+      expectedState[0].position = 1;
```

Comment:
I think the thing to be checking here is the key and/or the display name is now different (from x -> y). Possibly also that the keys of the activitySections are now c and d

Reply:
That should all be covered by the `deepEqual`, since the slice + setting the position has changed the `expectedState` into the correct state. We could add additional assertions after the `deepEqual` on key / display name though as a sanity check. I'm fine with it either way.

We could also try to write the logic for generating the `expectedState` in some other way if it'd be more clear.